Mage.Sets/src/mage/cards/r/RipplesOfPotential.java

```diff
@@ -0,0 +1,133 @@
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.permanent.PermanentIdPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.TargetPermanent;
+import mage.watchers.Watcher;
+
+/**
+ * Ripples of Potential {1}{U}
+ * Instant
+ * Proliferate, then choose any number of permanents you control that had a counter put on them this way. Those permanents phase out.
+ *
+ * @author DominionSpy
+ */
+public class RipplesOfPotential extends CardImpl {
+
+    public RipplesOfPotential(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{1}{U}");
+
+        // Proliferate, then choose any number of permanents you control that had a counter put on them this way. Those permanents phase out.
+        // A watcher is required as another effect may prevent counters from being put on permanents (e.g. Solemnity)
+        this.getSpellAbility().addEffect(new ProliferateEffect());
+        this.getSpellAbility().addEffect(new RipplesOfPotentialEffect());
+        this.getSpellAbility().addWatcher(new RipplesOfPotentialWatcher());
+    }
+
+    private RipplesOfPotential(final RipplesOfPotential card) {
+        super(card);
+    }
+
+    @Override
+    public RipplesOfPotential copy() {
+        return new RipplesOfPotential(this);
+    }
+}
+
+class RipplesOfPotentialEffect extends OneShotEffect {
+
+    public RipplesOfPotentialEffect() {
+        super(Outcome.Benefit);
+        staticText = "Proliferate, then choose any number of permanents you control that had a counter put on them this way. Those permanents phase out.";
+    }
+
+    private RipplesOfPotentialEffect(final RipplesOfPotentialEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public RipplesOfPotentialEffect copy() {
+        return new RipplesOfPotentialEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        RipplesOfPotentialWatcher watcher = game.getState().getWatcher(RipplesOfPotentialWatcher.class, source.getSourceId());
+        if (watcher != null) {
+            FilterPermanent filter = new FilterPermanent("permanents");
+            filter.add(Predicates.or(
+                    watcher
+                            .getProliferatedPermanents()
+                            .stream()
+                            .map(game::getPermanent)
+                            .filter(Objects::nonNull)
+                            .filter(permanent -> permanent.isControlledBy(source.getControllerId()))
+                            .map(Permanent::getId)
+                            .map(PermanentIdPredicate::new)
+                            .collect(Collectors.toSet())
```

Comment:
Note that you aren't checking zcc here, though it should be fine for this application since it occurs immediately after the proliferation. But in general zcc is important (e.g. checking MageObjectReference instead of just UUID) so that a flickered/recast/moved object isn't affected by things referring to its previous incarnation. See `PermanentReferenceInCollectionPredicate`.